### ISSUE TITLE
Profiles handling within LauncherFrame

### ISSUE THREAD

**Issue by MikeNeilson** [opendcs/opendcs#289]
**Component to be refactored**

LauncherFrame/Profiles

**Purpose of refactoring**

The current Profile system has two different but identical(ish) functions for loading the profiles.
Additionally the current code while it looks at the DCSTOOL_USERDIR assumes "decodes.properties" instead of "user.properties"
as the default profile path.

**Proposed method**

- Add type safety, ProcCallBack takes an Object instead of a Generic and then does conversions, this can cause unexpected and confusing class cast exceptions
  - Given the functional interfaces and futures of Java 8+ it also may no longer be necessary; if so new standards should be used
- Currently the LauncherFrame detects on button click if a non-default profile is used and goes down a different code path.
  - I think all "button clicks" should just go down the path of getting launched that way. This would reduce mental load while debugging and make it easier to say, put in a "fake" ProfileServerThread so testing can verify commands are sent correctly without actually having to start anything.
  - BasicServer is used, but this really doesn't need to use this style of IPC and should be investigated.

The short version is:
Instead of profileCombo being `JComboBox<String>` it will become `JComboBox<Profile>` and every button click can just get the current Profile and not have to worry about determining the correct path later, it would already be set.

The Profile class should get a static member `List<Profile>` getProfiles(Path searchDir)`

The Profile Server Thread would also just take this object and use it for lookup on whatever method it ends up using.

The model in the profile viewer/creator could then also be extended as the Profile Model can hold more data such as the database type.

**Additional context**
